grails-app/migrations/views/product-search.sql

```diff
@@ -0,0 +1,20 @@
+CREATE OR REPLACE VIEW product_search AS (
+SELECT
+  concat(p.id, ' ', p_a.location_id)            as id,
+  p.id                                          as product_id,
+  p_a.location_id                               as location_id,
+  p.product_type_id                             as type_id,
+  CASE
+  	WHEN GROUP_CONCAT(p_t_s_a.product_activity_code, ' ') is null THEN FALSE -- concat with ' ' for easier check, because other codes might include SEARCHABLE key word in code
```

Comment:
Ooh this might be pretty bad performance-wise since we won't be able to index this. We can give this a shot, but it might be possible to allow product search to be the explosion and the query can change the grain to product-location. In other words, we don't GROUP BY until we get to the query. Leave it as-is for now and we can investigate in another ticket.

Reply:
Quick check: I am getting results in ~600ms for a single location search (no product specified) in the database workbench. When I removed this property that is using `GROUP_CONCAT`, I am getting the same results in ~730ms. So having the `GROUP_CONCAT` improves the performance in this case somehow.